### ISSUE TITLE
expose log_inv_logit_diff function

### ISSUE THREAD

**Issue by andrjohns** [stan-dev/math#855]
#### Summary:
Function for computing the log of the difference between two inverse logits (log_inv_logit_diff) is declared within ```prim/mat/prob/ordered_logistic_lpmf.hpp```, rather than as a standalone function.

#### Description:
Declaring the function only within another distribution forces the gradient calculation to rely on autodiff and keeps it from wider use

#### Current Version:
v2.17.0

**Comment by bob-carpenter** [comment]
Thanks for posting.  I changed the title to correspond to what needs to be done.
